sections/video.liquid

```diff
@@ -100,22 +92,39 @@
         </span>
       </button>
       <template>
-        {%- if section.settings.video_url.type == 'youtube' -%}
-          <iframe
-            src="https://www.youtube.com/embed/{{ section.settings.video_url.id }}?enablejsapi=1"
-            class="js-youtube"
-            allow="autoplay; encrypted-media"
-            allowfullscreen
-            title="{{ section.settings.description | escape }}"
-          ></iframe>
+        {%- if section.settings.video == null -%}
+          {%- liquid
+            assign loop = ''
+            if section.settings.enable_video_looping
+              assign loop = '&loop=1&playlist=' | append: video_id
+            endif
+          -%}
+          {%- if section.settings.video_url.type == 'youtube' -%}
+            <iframe
+              src="https://www.youtube.com/embed/{{ video_id }}?enablejsapi=1&autoplay=1{{ loop }}"
+              class="js-youtube"
+              allow="autoplay; encrypted-media"
+              allowfullscreen
+              title="{{ section.settings.description | escape }}"
+            ></iframe>
+          {%- else -%}
+            <iframe
+              src="https://player.vimeo.com/video/{{ video_id }}?autoplay=1{{ loop }}"
+              class="js-vimeo"
+              allow="autoplay; encrypted-media"
+              allowfullscreen
+              title="{{ section.settings.description | escape }}"
+            ></iframe>
+          {%- endif -%}
         {%- else -%}
-          <iframe
-            src="https://player.vimeo.com/video/{{ section.settings.video_url.id }}"
-            class="js-vimeo"
-            allow="autoplay; encrypted-media"
-            allowfullscreen
-            title="{{ section.settings.description | escape }}"
-          ></iframe>
+          {{ section.settings.video
+            | video_tag:
+              image_size: "2048x",
+              autoplay: true,
```

Comment:
What does the deferred media behaviour get us in terms of performance for hosted video?

The deferred media pattern is great for youtube and vimeo since their video players are very large (lots of JS). For hosted video though I believe we are using the default browser controls. Do we need it for hosted video? It might solve some problems with the flicker and double-clicking play on Safari.

Reply:
> The deferred media pattern is great for youtube and vimeo since their video players are very large

Correct. Using deferred media was mostly to take advantage of existing functionality and styling. If I were to not defer the rendering, I'd also use `preload="metadata"` on the video to avoid any flicker. This will load that poster/fallback image I can't exclude from the video_tag and it will preload a certain amount of the video itself so it can at least show the first frame (usually ~1kb but sometimes it just starts buffering for some reason). So inherently there will be a slight performance hit that deferred media aims to avoid. If I don't preload anything, the flicker will occur regardless.

@melissaperreault it could be worth us pairing on this and considering the severity and our options. We could also just mitigate the flicker as much as I can for now and follow up with an enhancement if we feel we need to. Youtube and vimeo also flash a blank background before the first frame loads.

Reply:
Paired with @melissaperreault on this. We decided the flicker isn't blocking, but we did set background of the video player to pure black. We find this slightly less jarring and should be a familiar experience to the youtube video loading sequence.

I experimented with non-deferred loading but didn't feel the benefit was worth the performance impact or the code exceptions. 

I also tried changing the z-index of the cover image instead of hiding it. This actually worked well, but only if the aspect ratio of the video and cover image were the same.

I will likely be opening an issue for follow up investigations/improvements as a result of this PR and will make mention all this.